### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by jojo987N**
### To Do list: add & remove
For this milestone :

-  Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task.
- Implement a function for deleting a task
- Implement a function for editing task descriptions.
- All changes to the To Do List is saved in local storage.
- Make sure that there are no linter errors

From the demo, this is how the remove functionality works. You must first delete the text, then click on the delete icon. I reproduced exactly  the project guideline.
https://www.youtube.com/watch?v=AcUd-_Yjjqg&ab_channel=EmiliaAndrzejewska
https://web.archive.org/web/20180320194056/http://www.getminimalist.com:80/

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -49,4 +30,62 @@ const listFooter = `</ul>
...
+      icons[0].classList.add('hide');
+      icons[1].classList.remove('hide');
+
+      icons[1].onclick = () => {
+        if (!input.value) {
+          data = removeTask(i, data);
+
+          display();
+        }
+      };
```
Comment (AmaduKamara) [inline src/index.js:59]: - [ ] The remove functionality is not working as expected when the delete icon is clicked. Kindly work on it so a todo is removed when the delete icon is clicked 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @jojo987N,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Add and edit todo features work fine as expected 👍🏽
✅ The app is nicely styled 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly work on the remove functionality so a todo is removed successfully 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js (hunk 1 above)
Comment (jojo987N) [reply src/index.js:59]: From the demo, this is how the remove button works. You must first delete the text, then click on the delete button. I reproduced exactly what the demo and the site do.
https://www.youtube.com/watch?v=AcUd-_Yjjqg&ab_channel=EmiliaAndrzejewska
https://web.archive.org/web/20180320194056/http://www.getminimalist.com:80/

File: src/index.js [hunk 2]
```diff
@@ -49,4 +30,62 @@ const listFooter = `</ul>
...
+
+      e.target.onkeypress = (event) => {
```
Comment (codecaiine) [inline src/index.js:61]: - [ ] [ OPTIONAL] Good job 👏  by implementing the deleting function as on the video shared above ⬆️  but I would like you to consider the fact that we can delete the task when the delete icon is  checked As mentioned by the previous reviewer 👍

File: src/index.js [hunk 3]
```diff
@@ -49,4 +30,62 @@ const listFooter = `</ul>
...
+  });
+  const enter = document.querySelector('.enter');
+  enter.onkeypress = (e) => {
+    if (e.key === 'Enter') {
+      if (!enter.value) return;
+
+      addNewTask({
+        description: enter.value,
+        completed: false,
+        index: data.length,
```
Comment (codecaiine) [inline src/index.js:81]: - [ ] [OPTIONAL] According to the project requirement by default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5) . Please fix it to make your project works as expected 👍

**Review by codecaiine — APPROVED** [review]
Hi @jojo987N , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Adding , editing and deleting functions are well implemented ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**